src/resolve-config-editorconfig.js

```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+const editorconfig = require("editorconfig");
+const mem = require("mem");
+const pathRoot = require("path-root");
+
+const maybeParse = (filePath, config, parse) => {
+  const root = filePath && pathRoot(filePath);
+  return filePath && !config && parse(filePath, { root });
+};
+
+const editorconfigAsyncNoCache = (filePath, config) => {
+  return Promise.resolve(maybeParse(filePath, config, editorconfig.parse)).then(
+    editorConfigToPrettier
+  );
+};
+const editorconfigAsyncWithCache = mem(editorconfigAsyncNoCache);
+
+const editorconfigSyncNoCache = (filePath, config) => {
+  return editorConfigToPrettier(
+    maybeParse(filePath, config, editorconfig.parseSync)
+  );
+};
+const editorconfigSyncWithCache = mem(editorconfigSyncNoCache);
+
+function getLoadFunction(opts) {
+  if (opts.sync) {
+    return opts.cache ? editorconfigSyncWithCache : editorconfigSyncNoCache;
+  }
+
+  return opts.cache ? editorconfigAsyncWithCache : editorconfigAsyncNoCache;
+}
+
+function clearCache() {
+  mem.clear(editorconfigSyncWithCache);
+  mem.clear(editorconfigAsyncWithCache);
+}
+
+function editorConfigToPrettier(editorConfig) {
+  editorConfig = editorConfig || {};
+  const result = {};
+
+  if (editorConfig.indent_style) {
+    result.useTabs = editorConfig.indent_style === "tab";
+  }
+
+  const tabWidth = editorConfig.indent_size || editorConfig.tab_width;
```

Comment:
It's an edge case, but technically I believe it's possible to define both an `indent_size` and a `tab_width`, where the latter should be preferred in case `indent_style` is `tab`. In fact, `indent_size` can actually be set to `tab`. This line would, therefore, get the wrong prettier `tabWidth` or break completely?

https://github.com/prettier/prettier-atom/blob/master/src/executePrettier/buildEditorConfigOptions.test.js#L15-L21